src/Pos/Generator/Block/Mode.hs

```diff
@@ -165,6 +177,14 @@ mkBlockGenContext bgcParams = do
         bgcTxpMem <- (,ignoreTxpMetrics) <$> mkTxpLocalData
         bgcDelegation <- mkDelegationVar @SscGodTossing
         return BlockGenContext {..}
+  where
+    bgcCustomUtxo =
+        let addrs = map (makePubKeyAddress . toPublic) . toList $
+                    view (bgpSecrets . asSecretKeys) bgcParams
+            utxoTxHash = unsafeHash ("randomutxotx" :: Text)
+            txIns = map (TxIn utxoTxHash) [0..fromIntegral (length addrs) - 1]
+            txOuts = map (\addr -> TxOutAux (TxOut addr (mkCoin 10000)) []) addrs
+        in M.fromList $ txIns `zip` txOuts
```

Comment:
1. Wtf is that? Why do you create utxo here? Block generator definitely shouldn't define genesis utxo by itself. It doesn't know which utxo was genesis! It should come from the outside!
2. Why does it have empty distribution? It doesn't make sense.
3. Actually I would remove genesis utxo and store only genesis stakeholders (and change constraints is some places to make it possible).
4. It seems that here you can just use current utxo and set it as genesis utxo. If we are in bootstrap era, it's guaranteed that all stake always belongs to all genesis stakeholders and only them. If we are not in bootstrap era, we don't care about genesis stakeholders. 

Anyway, the ideal solution for me is to change `HasLens GenesisUtxo` to `HasLens GenesisStakeholders` in some places, store only `GenesisStakeholders` in `BlockGenContext` and pass it here through `MonadBlockGen`.